source/CppTestSrv/SERVER.IDL

```diff
@@ -0,0 +1,72 @@
+//
+// Server.idl
+//
+// This file will be processed by the MIDL compiler to
+// produce the type library (server.tlb) and marshaling code.
+//
+
+import "oaidl.idl" ;
+
+typedef [uuid(00FABB0F-5691-41A6-B7C1-11606671F8E5)]
+struct T_TEST_RECORD {
+	BSTR question ;
+	long answer ;
+	VARIANT_BOOL needs_clarification ;
+} T_TEST_RECORD ;
+
+
+// Interface IDualRecordParamTest
+[
+	odl,
+	uuid(0C4E01E8-4625-46A2-BC4C-2E889A8DBBD6),
+	dual,
+	helpstring("Dual Interface for testing record parameters."),
+	nonextensible,
+	oleautomation
+]
+interface IDualRecordParamTest : IDispatch
+{
+	[id(0x00000001)]
+	HRESULT InitRecord([in, out] T_TEST_RECORD* test_record) ;
+	[id(0x00000002)]
+	HRESULT VerifyRecord(
+					[in] T_TEST_RECORD* test_record,
+					[out, retval] VARIANT_BOOL* result);
+} ;
+
+
+// Interface IDispRecordParamTest
+[
+	uuid(033E4C10-0A7F-4E93-8377-499AD4B6583A),
+	helpstring("Dispinterface for testing record parameters.")
+]
+dispinterface IDispRecordParamTest
+{
+	interface IDualRecordParamTest;
+} ;
+
+
+//
+// Component and type library descriptions
+//
+[
+	uuid(07D2AEE5-1DF8-4D2C-953A-554ADFD25F99),
+	version(1.0),
+	helpstring("Comtypes C++ Test COM Server 1.0 Type Library.")
+]
+library ComtypesCppTestSrvLib
+{
+	// TLib : OLE Automation : {00020430-0000-0000-C000-000000000046}
+	importlib("stdole2.tlb") ;
+
+	// CoComtypesDispIfcParamTests
+	[
+		uuid(06571915-2431-4CA3-9C01-53002B060DAB),
+		helpstring("Comtypes component for dispinterface parameter tests.")
+	]
+	coclass CoComtypesDispIfcParamTests
+	{
+		interface IDualRecordParamTest ;
+		dispinterface IDispRecordParamTest ;
+	} ;
```

Comment:
I think it would be better to comment on what kind of interfaces should be (or should not be) implemented (added) in this component in the future.